guvnor-webapp/src/main/java/org/guvnor/client/perspectives/M2RepoPerspective.java

```diff
@@ -61,23 +58,16 @@ private void init() {
         add( m2RepoEditor );
     }
 
+    @Inject
+    UploadFormPresenter uploadFormPresenter;
+
     @WorkbenchMenu
     public Menus getMenus() {
         return MenuFactory.newTopLevelMenu( AppConstants.INSTANCE.Upload() )
                 .respondsWith( new Command() {
                     @Override
                     public void execute() {
-                        final UploadForm uploadForm = iocManager.lookupBean( UploadForm.class ).getInstance();
-                        //When pop-up is closed destroy bean to avoid memory leak
-                        uploadForm.addCloseHandler( new CloseHandler<PopupPanel>() {
-
-                            @Override
-                            public void onClose( CloseEvent<PopupPanel> event ) {
-                                iocManager.destroyBean( uploadForm );
-                            }
-
-                        } );
-                        uploadForm.show();
+                        uploadFormPresenter.showView();
                     }
```

Comment:
Is it correct realization? Can I do not destroy it? If yes I can't to clear Filename field because there is no method in FileUpload form. So it can to be always with previous value or we have to add setFilename method to org.uberfire.ext.widgets.common.client.common.FileUpload or at least have to destroy this uploadFormView and recreate it again. What is the easiest way to solve it?